### DISCUSSION TITLE
Citations & Bibliography [Zotero-Obsidian-Rmarkdown] [Complete workflow]

### DISCUSSION THREAD

**Discussion by jcmoral90** [BryanJenksCommunity/FAQ#47] [Q&A] [answered]
Hello everyone!

First I would like to thank Bryan and Obsidian users like Bri Watson and Argentum for their contribution in developing a satisfactory workflow for the research. 

Before I ask my questions, I would like to make a brief description of the steps (obtained from the multiple videos already uploaded on youtube or discussions on the forum).

**In Zotero...** 

1. Extracting notes with Zotfile and Mdnotes in Zotero
2. Modify Zotfile preferences to give color names of the annotations . E.g **magenta** **red**, associated to different themes. In my case Problems and Discussions (red), Methodology (Green), State of the Art (Yellow), Numerical Data (Blue).
3. Extract tables from pdfs with Tabula app, to export to JSON and transform to Markdown.
4. Export notes by paper with Md notes (In my case I use the single File or Batch).
5. In mdnotes or zotfile preferences rename the file as @citekey, this way I will have a @citekey.md exported in my Obsidian vault.

**In Obsidian...**

5. To each color with the obsidian replace tool (I think Ctrl + Shift + H if I am not wrong) I replace the colors with tags like #ResearchProblem so I can search them in the future, and also add subtags like #ResearchProblem/ArtificialIntelligence.
6. Then I reread my notes and assign new tags. 
7. I re-read my notes and write down topics for example from a paper [[1. Introduction]], [[2. Results]], etc 
8. Remember that our annotations are as @citekey, so if I want to quote a note in these chapters e.g [[1. Introduction]] Introduction]], then it will appear as [[@citekey]].

**RMarkdown...(?)**

Let's imagine I have written a paper, citing with [[@citekey]]. I would only need to remove one bracket in order to leave the citations as [@citekey] so that pandoc can process them in Rmarkdown. At the moment the only thing I can think of is to delete the remaining brackets with the replace tool in Obsidian and just copy-paste them into RMarkdown.

This way we could complete a workflow from the literature review, the use of Obsidian as a Second Brain, idea generator and writer, and finally RMarkdown to see only the format of a scientific paper or report.

I read attentively your suggestions to improve the workflow

-Greetings from Chile

**Comment by tallguyjenks** [comment, marked as answer]
one way to automate the bracket removal for the `[[@citekey]]` in the RMarkdown document would be to write a simple shell script to find and replace any of those cite keys with a single square bracket instance. 

would really only need the `sed` command and use some regular expressions
